tests/unit/test_try.py

```diff
@@ -28,3 +28,73 @@
 )
 def test_options_arch(params: dict[str, Any], expected: dict[str, Any]):
     assert tmt.cli.trying._construct_trying_provision_options(params) == expected
+
+
+@pytest.fixture
+def mock_try_execution():
+    """Fixture to mock Try execution and capture CLI invocation"""
+    from unittest.mock import patch
+
+    with (
+        patch('tmt.trying.Try.go') as mock_go,
+        patch('tmt.trying.Try.store_cli_invocation') as mock_store,
+    ):
+        yield mock_go, mock_store
+
+
+@pytest.fixture
+def cli_runner():
+    """Fixture providing a Click CLI runner"""
+    from click.testing import CliRunner
+
+    return CliRunner()
+
+
+@pytest.mark.parametrize(
+    ('env_args', 'expected_env_vars'),
+    [
+        # Single environment variable
+        (['--environment', 'TEST_VAR=test_value'], ['TEST_VAR=test_value']),
+        # Multiple environment variables with different flags
+        (
+            ['--environment', 'VAR1=value1', '--environment', 'VAR2=value2', '-e', 'VAR3=value3'],
+            ['VAR1=value1', 'VAR2=value2', 'VAR3=value3'],
+        ),
+        # Environment file syntax
+        (['--environment', '@/path/to/env.yaml'], ['@/path/to/env.yaml']),
+        # Mixed variable and file syntax
+        (
+            ['--environment', 'DEBUG=1', '-e', '@/path/to/secrets.env'],
+            ['DEBUG=1', '@/path/to/secrets.env'],
+        ),
+    ],
+)
+def test_try_command_processes_environment_options(
+    cli_runner, mock_try_execution, env_args, expected_env_vars
```

Comment:
The environments should be generalized since it is something we test across multiple tests